### ISSUE TITLE
Curate a list of open source projects

### ISSUE THREAD

**Issue by avsm** [ocaml/v2.ocaml.org#497]
It's been noted on this blog that "good open source OCaml projects are still somewhat hard to come by in general"; http://blog.nullspace.io/beginners-guide-to-ocaml-beginners-guides.html

It would be nice to be able to point people to a list of substantive open-source codebases to browse through.  Off the top of my head:
- mldonkey
- lwt / ocsigen
- coq
- frama-c
- mirage
- pfff
- xapi
- cil
- batteries
- core
- liquidsoap

(and no doubt many more that we can cherry pick from the opam database)

**Comment by amirmc** [comment]
Good idea. In general this theme seems to crop up every now and again (reminds me of a conversation w @lpw25 last year). 

In any case, this prompts me to think of the 'success stories' page and whether what you've described is a different (& better) form of that page. 

Best wishes,
Amir

## 

sent via mobile

**Comment by c-cube** [comment]
I'd add [why3](http://why3.lri.fr) too.

**Comment by avsm** [comment]
- libguestfs : https://github.com/libguestfs/libguestfs
- goaljobs: http://people.redhat.com/rjones/goaljobs/

(these are nice, simple unix utilities).

**Comment by amirmc** [comment]
I'm guessing this was accidentally closed. Re-opening.

**Comment by edwintorok** [comment]
I highly recommend reading through the documentation written by @dbuenzli : http://erratique.ch/software. 
It doesn't only provide reference docs for the APIs, but also gives a short introduction/context, documents conventions and provides short usage examples, etc. Also the libraries themselves should be quite easy to use.

OCamlNet is also worth looking at, if you keep in mind that its design decisions were taken quite some time ago (and not necessarily how a brand new library would be designed today).

**Comment by Chris00** [comment]
Maybe we should aim to provide a recommended path (or at least pointers) to learn OCaml, reading the source of a list of projects being only one of them.  An idea is to ditch the [code examples](http://ocaml.org/learn/) section and replace it with a bullet list of recommendations (the links to the code examples, pleac,... being in this list).

**Comment by ghost** [comment]
I started a section in awesome-ocaml to list interesting source code. I can add these additional links there soon. That might be sufficient to close this issue?

**Comment by ghost** [comment]
https://github.com/ocaml-community/awesome-ocaml/pull/194

I am not sure about batteries, lwt. I wouldn't steer new people to read code that has to be very clever. I haven't looked at their source code yet, but do you all consider the code complex/clever or is it more lots of simple code that beginners can grok?

goaljobs seems inactive, didn't add it

I'll try pick off items from dbuenzli's list and then close this after creating a second awesome-ocaml PR with those projects.
